### PR TITLE
vdrift: migrate to scons_latest

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

#201859

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/vdrift/default.nix [hunk 1]
```diff
@@ -33,10 +34,22 @@ let
...
+      (fetchpatch {
+        name = "scons-python-3-fixes.patch";
+        url = "https://github.com/VDrift/vdrift/commit/2f1f72f2a7ce992b0aad30dc55509b966d1bb63d.patch";
+        sha256 = "sha256-gpIB95b1s+wpThbNMFXyftBPXkZs9SIjuCcvt068uR8=";
+      })
+      (fetchpatch {
+        name = "sconstruct-python-3-fix.patch";
+        url = "https://github.com/VDrift/vdrift/commit/7d04c723a165109e015204642f4984a1a4452ccb.patch";
+        sha256 = "sha256-ASEV46HnR90HXqI9SgHmkH2bPy5Y+vWN57vEN4hJMts=";
+      })
```
Comment (thiagokokada) [inline pkgs/games/vdrift/default.nix:51]: Shouldn't we just bump it to the latest version instead?

Comment (wegank) [reply pkgs/games/vdrift/default.nix:51]: The latest version crashes on my machine, so I prefer to leave the work to the maintainer.

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 209320` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vdrift-bin</li>
  </ul>
</details>

**Comment by thiagokokada** [comment]
Tested and works fine.

**Pull request merged**